MetasysServices/MetasysClient.cs

```diff
@@ -1207,7 +1209,58 @@ private async Task SendCommandRequestAsync(Guid id, string command, IEnumerable<
             }
         }
 
+        #region "ad-hoc calls" // =========================================================================================================
+        // Support Ad-Hoc calls -----------------------------------------------------------------------------------------------------------
+        ///<inheritdoc/>
+        public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage requestMessage, HttpCompletionOption completionOption = HttpCompletionOption.ResponseContentRead, CancellationToken cancellationToken = default)
+        {
+            var response = new HttpResponseMessage();
+            try
+            {
+                var baseUri = new Uri(Client.BaseUrl);
+
+                if (!baseUri.IsBaseOf(requestMessage.RequestUri))
+                {
+                    //By default it failed with 401 unauthorized. Do we need to handle it explicitly
+                    //Requesting to connect different host.
+                }

```

Comment:
`IsBaseOf` is an interesting test. Generally, it's a "starts with" check, but technically a little more complicated.

First, I don't think we should support connecting to a different host. Who cares? Someone can just write their own HTTP client for that.

Secondly, we should support arbitrary API versions. And I don't think `IsBaseOf` would detect that with how `Client` is currently configured (the selected API version becomes the base URI of the client in `InitFlurlClient`).

So it's probably better to use other properties or methods of the `Uri` class, like the `Compare` function with maybe `SchemeAndServer`?